src/video/wayland/SDL_waylandmouse.c

```diff
@@ -403,7 +416,21 @@ Wayland_ShowCursor(SDL_Cursor *cursor)
 static void
 Wayland_WarpMouse(SDL_Window *window, int x, int y)
 {
-    SDL_Unsupported();
+    SDL_VideoDevice *vd = SDL_GetVideoDevice();
+    SDL_VideoData *d = vd->driverdata;
+    struct SDL_WaylandInput *input = d->input;
+
+    if (input->cursor_visible == SDL_TRUE) {
+        SDL_Unsupported();
+    } else if (!SDL_GetHintBoolean(SDL_HINT_VIDEO_WAYLAND_EMULATE_MOUSE_WARP,
```

Comment:
This shouldn't check the hint every time a warp happens. Instead you should check the value at initialization and respond to hint changes via callback. See the handling for SDL_HINT_MOUSE_RELATIVE_WARP_MOTION for an example of this.